tripy/tests/integration/test_functional.py

```diff
@@ -166,11 +166,11 @@ def test_weights_loading_from_torch(self, kind):
 
             tripy_linear = tp.Linear(2, 3)
             if kind == "gpu":
-                tripy_linear.weight = tp.Parameter(tp.Tensor(torch_linear.weight.to("cuda"), device=tp.device(kind)))
-                tripy_linear.bias = tp.Parameter(tp.Tensor(torch_linear.bias.to("cuda"), device=tp.device(kind)))
+                tripy_linear.weight = tp.Parameter(tp.Tensor(torch_linear.weight.detach().to("cuda"), device=tp.device(kind)))
```

Comment:
What happens when you remove `detach()`?

Reply:
Torch tensors will not allow invocation of the `__dlpack__` method if they have a `grad` tensor attached.

Reply:
can we instead handle this internally? This could lead to bad UX.

Reply:
Are you suggesting we add a check within the `Array._memref` function to see if the `data`  is a torch tensor and then manually calling `detach()`?

Reply:
Yeah, that would be better.